Improve split doc (#482)
KMKfw/kmk_firmware: docs/split_keyboards.md

side = SplitSide.RIGHT if str(getmount('/').label)[-1] == 'R' else SplitSide.LEFT

if side == SplitSide.RIGHT:
    keyboard.col_pins = ...
    keyboard.row_pins = ...
    keyboard.diode_orientation = ...
else:
    keyboard.col_pins = ...
    keyboard.row_pins = ...
    keyboard.diode_orientation = ...

split = Split(
    split_side=side,
    target_left=True,
    ...
)
```

Note: It is best to stay as coherent as possible, but thanks to the `coord_mapping` feature, none of the `col_pins`, `row_pins` and `diode_orientation` **need** to be the same for both side. The only need is for `len(col_pins) * len(row_pins)` to be the same (as it is assumed that both side will handle the same numbers of potential keys.

#### `split_flip`
If your split keyboard use the **same pcb for both side**, but vertically flipped, Set this to `True`, and `False` otherwise. `True` will means everything are the same for both side except that `row_pins` are reversed (as you flip one of the pcb verticaly).

You will need to update your `coord_mapping` (if you use one) if you change this value.

#### `split_target_left`
The `split_target` refer to the side that will send keypress over usb.

> If you use EE HANDS (`split_side = None`) this parameter will be automatically overriden (a side connected to a usb host will act as a `split_target`).
> 
> If setting this manually, be sure to reflect the actual usb connection and make sure to use the same value for both sides !

Reviewer: ```suggestion
Setting `split_side = None` (similar to EE HANDS in QMK) this parameter will be overriden.
```
* KMK doesn't have EE HANDS. It's a similar feature that just copied the name from QMK.
* I removed some redundant text.

#### `uart_flip`
If your boards are connected through the **same** pins (like gpio**4** of board A to gpio**4** of board B): use `uart_flip = True`.

If your boards are connected through **different** pins (like gpio**4** of board A to gpio**10** of board B): use `uart_flip = False`.

For example if using the qwiic/stemma connector of the [KB2040](https://www.adafruit.com/product/5302) (supposed to be for I2C) for the uart connection. You will need to use `uart_flip = True` as for i2c sda connect to sda and scl to scl.

> 

Author: I removed the example to avoid confusion.

#### `use_pio`
If you're using RP2040 based board and want to use other pins that the one with hardware support of RX/TX, you can try using PIO implementation of two-way wire protocol. Typical use cases for it are premade boards, made with QMK's bitbanging protocols in mind.

In order to enable it, you must:

- Install Circuit Python in 7.2+ version
- Add `adafruit_pioasm.mpy` library to lib or root folder of a board
- Pass `use_pio=True` into `Split()` constructor

> The file is available [here](https://github.com/adafruit/Adafruit_CircuitPython_PIOASM/releases). From the `adafruit-circuitpython-pioasm-x.y-mpy-0.x.y.zip`, copy the `lib` folder (or its content) to the root of your circuitpython drives. Make sure to get the zip corresponding to your circuitpython version as the mpy format can evolve!
> 
> Continuing our example of the KB2040, one side's UART being fliped will need to use a pin capable of only tx as rx (and vice-versa). So in order to use that qwiic/stemma connector you will also need `use_pio = True`.

Reviewer: ```suggestion
The file is available [here](https://github.com/adafruit/Adafruit_CircuitPython_PIOASM/releases). From the `adafruit-circuitpython-pioasm-x.y-mpy-0.x.y.zip`, copy the `lib` folder (or its content) to the root of your circuitpython drives. Make sure to get the zip corresponding to your circuitpython version as the mpy format can evolve!

Continuing our example of the KB2040, one side's UART being fliped will need to use a pin capable of only tx as rx (and vice-versa). So in order to use that qwiic/stemma connector you will also need `use_pio = True`.
```
* Please don't use "[here]" as a name for a link. Give it the name or description it's pointing at.
* The second paragraph is very confusing again. When did we get from RP2040 to KB2040? What does "use a pin capable of only tx as rx" mean? Why are we talking about qwicc/stemma suddenly? Sorry, but this is all very incoherent.

#### `data_pin`/`data_pin2`

> For UART `SplitType`: on the master or `split_target` side, `data_pin` is the one use for RX, `data_pin2` the one for TX.

Reviewer 2: We don't use master/slave naming schema. Please use "target" in term of "master" and "initiator" in term of "slave" at any point in docs/code. If that has slipped into the docs/code, it's a bug, and should be changed to the intended terminology.

    split_flip=True,  # If both halves are the same, but flipped, set this True
    split_side=None,  # Sets if this is to SplitSide.LEFT or SplitSide.RIGHT, or use EE hands
    split_type=SplitType.UART,  # Defaults to UART
    split_target_left=True,  # Assumes that left will be the one on USB. Set to False if it will be the right
    uart_interval=20,  # Sets the uarts delay. Lower numbers draw more power
    data_pin=None,  # The primary data pin to talk to the secondary device with
    data_pin2=None,  # Second uart pin to allow 2 way communication
    uart_flip=True,  # Reverses the RX and TX pins if both are provided
    use_pio=False,  # Use RP2040 PIO implementation of UART. Required if you want to use other pins than RX/TX
)

```

#### `split_side`
This tells your micro wich side it handles. For simple case, it is not needed as `split_side = None` will imply:
- auto dectection of the side from the drive name (ending with 'R'/'L').
- The `split_target` will be overriden. Each side will act as a `split_target` if connected to a usb host.

This is a sane default settings.

But you still can choose to set all that manually. Some reasons would be:

> - easy both side debug and upload by having both halves connected to usb (and uart between them). Explicit `split_side` and only one `split_target` is needed to avoid each half acting as a `split_target` (duplicating each keypress).

Reviewer 2: Scentences start with a capital letter. `easy` should be `Easy` as an example. This happens many places. Please check them all that were added to keep the docs as consistent as possible.

- wanting to be explicit about hardware discrepancy between halves (others than just reversing the rows see `split_flip`):

```python
from storage import getmount

side = SplitSide.RIGHT if str(getmount('/').label)[-1] == 'R' else SplitSide.LEFT

if side == SplitSide.RIGHT:

>     keyboard.col_pins = ...

Reviewer 2: This is only needed if PCB's are not interchangeable between sides, which is the common practice. maybe note that this is optional if they have different pinouts.

- install Circuit Python in 7.2+ version
- add `adafruit_pioasm.mpy` library to lib or root folder of a board
- pass `use_pio=True` into `Split()` constructor

The file is available [here](https://github.com/adafruit/Adafruit_CircuitPython_PIOASM/releases). From the `adafruit-circuitpython-pioasm-x.y-mpy-0.x.y.zip`, copy the `lib` folder (or its content) to the root of your circuitpython drives. Make sure to get the zip corresponding to your circuitpython version as the mpy format can evolve!

Continuing our example of the KB2040, one side's UART being fliped will need to use a pin capable of only tx as rx (and vice-versa). So in order to use that qwiic/stemma connector you will also need `use_pio = True`.

#### `data_pin`/`data_pin2`
For UART `SplitType`: on the master or `split_target` side, `data_pin` is the one use for RX, `data_pin2` the one for TX.

> ### EE HANDS

Reviewer 2: Because this name is confusing for some, maybe we change this to `### EE HANDS/AUTO HANDEDNESS`
